src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Filters/LiquidViewFilters.cs

```diff
@@ -107,7 +109,8 @@ static async ValueTask<FluidValue> Awaited(Task<IHtmlContent> task)
                 StringValue value;
                 using (var writer = new StringWriter())
                 {
-                    task.Result.WriteTo(writer, NullHtmlEncoder.Default);
+                    var htmlEncoder = ShellScope.Services.GetRequiredService<HtmlEncoder>();
```

Comment:
You should use the AmbientValues of the context to get a service. Not the ShellScope.

Look at the ShortCodeFilter for an example.

Reply:
> You should use the AmbientValues of the context to get a service. Not the ShellScope.

Is there any difference?

Reply:
I am not aware what the difference is, but it removes the need to depend on the ShellScope. I can only assume it allows for use in different contexts and also better testing (without having to implement a shell scope).